gulpfile.js

```diff
@@ -21,18 +21,19 @@ const flat = require('flat');
 const { argv } = require('yargs');
 const os = require('os');
 const { ExtensionRootDir } = require('./build/util');
-
 const isCI = process.env.TF_BUILD !== undefined || process.env.GITHUB_ACTIONS === 'true';
 
-gulp.task('compile', (done) => {
-    let failed = false;
-    const tsProject = ts.createProject('tsconfig.json');
-    tsProject
-        .src()
-        .pipe(tsProject())
-        .on('error', () => (failed = true))
-        .js.pipe(gulp.dest('out'))
-        .on('finish', () => (failed ? done(new Error('TypeScript compilation errors')) : done()));
+gulp.task('compile', async (done) => {
+    // Use tsc so we can generate source maps that look just like tsc does (gulp-sourcemap does not generate them the same way)
+    try {
+        const stdout = await spawnAsync('tsc', ['-p', './'], {}, true);
```

Comment:
This is necessary so we end up with the same output we would if we built locally.

Reply:
And the source maps are correct. Gulp-sourcemaps had a bunch of issues (couldn't rewrite sourceroot)